pkg/restore/crd.go

```diff
@@ -0,0 +1,209 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package restore
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rook/kubectl-rook-ceph/pkg/debug"
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func RestoreCrd(ctx context.Context, k8sclientset *k8sutil.Clientsets, operatorNamespace, clusterNamespace string, args []string) {
+	crd := args[0]
+	logging.Info("Detecting which ceph crd resources to restore")
+	getCrName := `kubectl -n %s get %s -o jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.metadata.deletionGracePeriodSeconds}{"\n"}{end}' | awk '$2=="0" {print $1}'`
+	command := fmt.Sprintf(getCrName, clusterNamespace, crd)
+	crName := strings.TrimSpace(exec.ExecuteBashCommand(command))
+
+	if crName == "" {
+		logging.Info("Nothing to do here, no crd resources with deletionGracePeriodSeconds 0")
+		return
+	}
+
+	if len(args) == 2 {
+		crName = args[1]
+	}
+
+	logging.Info("Restoring crd %s name %s", crd, crName)
+
+	logging.Info("Scaling down the operator to 0")
+	err := debug.SetDeploymentScale(ctx, k8sclientset.Kube, operatorNamespace, "rook-ceph-operator", 0)
+	if err != nil {
+		logging.Fatal(err)
+	}
+
+	logging.Info("Backing up kubernetes and crd resources")
+	getCrYamlContent := `kubectl -n %s get %s %s -oyaml > %s.yaml`
```

Comment:
```suggestion
	getCrYamlContent := `kubectl -n %s get %s %s -oyaml > %s.yaml`
```

This command will create the crd file which is used to re-create the cr later. Currently, this file is left there not removed after re-creating the cr. should we remove it after successful cr creation or leave the file as it is?

Reply:
If the restore is successful, seems like we should deleted it. It's really a temporary file. But if there is any failure, we should not delete it.

Reply:
Let's have a more unique filename that is the `crd-crname.yaml`, in case there are multiple CRs to restore.
```suggestion
	getCrYamlContent := `kubectl -n %s get %s %s -oyaml > %s-%s.yaml`
```